### DISCUSSION TITLE
QUESTION: How to do keydown event on a btn?

### DISCUSSION THREAD

**Discussion by liketoeatcheese** [gnat/surreal#31] [General]
Hey mate,

First of all, what a genius plugin!!!

I'm trying to get rid of a bunch of my js files with Surreal. How would I do me().on("multipleevents?"), like click and keydown (Esc) at the same time? For example, how would I achieve something like this but keep the locality of behaviour effect?
```html
<button id="cancelBtnDrawerForm"/>
```
```javascript
const cancelBtnDrawerForm = document.getElementById('cancelBtnDrawerForm')
const cancelDrawerForm = () => {
  form.classList.remove('show')
  bodyWrapper.classList.remove('scaled')
  body.classList.remove('scaled')
  setTimeout(() => {
    form.style.display = 'none'
    backdrop.style.display = 'none'
  }, 200)
}

cancelBtnDrawerForm.addEventListener('click', cancelDrawerForm)
document.addEventListener('keydown', (event) => {
  if (event.key === 'Escape') {
    cancelDrawerForm();
  }
});
```

**Comment by gnat** [comment]
Hey man, thank you for the kind words!

There's a lot of value in having threads like this to discuss patterns as people discover them, so I'll convert this to an ongoing discussion.

Handling multiple events at once is a challenge in Javascript in general- I'd love to hear what people come up with as well!
